### ISSUE TITLE
In high contrast aquatic or desert modes, cursor is not visible at 'plain text': A11y_MonacoEditor_High Contrast

### ISSUE THREAD

**Issue by AnKushSingh05** [microsoft/monaco-editor#4985]
### Monaco Editor Playground Link

https://microsoft.github.io/monaco-editor/playground.html?source=v0.40.0-dev.20230707#XQAAAAKlAgAAAAAAAABBqQkHQ5NjdMjwa-jY7SIQ9S7DNlzs5W-mwj0fe1ZCDRFc9ws9XQE0SJE1jc2VKxhaLFIw9vEWSxW3yscw2Kir5Juetpg8d5BCIdogK6CBmgyKHIOVjH6uO-oA-RA-rcpQdGBlgwMT7NDeGjyzFr76MYz0c_YZhkfVact66KR8EKI9H0wOi-Q6tpDt-O42AX4Vj2BBGCeVYFhuwwQHQ084HXevxFU6s-aiyWw_iVEKmIVfS1mf-DaP5DuuJ9ninHo2zYoxRUK6gbOmW-kkEFhe84sCJio9lqPwjgEUbfAMk52muZB43edii8a6ay0CNuV1k16w8W6kt0kIlpz_rHCLlPrBGxJ3G5aWhuvMqA05A2XT_oDw060WX7VLyNNhagA1yGSkMrHmbmj7PTZb-kmmUw

### GitHubTags:
#A11yTCS; #A11ySev2; #A11yMAS; #WCAG4.3.1; #Desktopweb; #Win11; #Visual Studio Code Client; #VisualStudioCode-Web-Apr2021; #ChromiumEdge; #a11y-monacoeditor; #DevTools; #PartnerBug; # No Disruption of Accessibility Features:#External:BUG6023023;CAI:BUG6023023

### Environment Details:
#URL: https://microsoft.github.io/monaco-editor/playground.html?source=v0.52.2#XQAAAAJjAQAAAAAAAABBqQkHQ5NjdMjwa-jY7SIQ9S7DNlzs5W-mwj0fe1ZCDRFc9ws9XQE0SJE1jc2VKxhaLFIw9vEWSxW3yscwzZypW_1uhs0-3YGTBOsVJ4S4FwfCDQb5-2uUHXTVAefbVSd9IzaZ_vv9VR9KugsKimARHSWlrTPOQHROITHxmAetfeRrm3VHdSe7_Etv6WpgzlH4o4eQ6dMnzwfwYcUqOiZ_ge3TaAC-S-3X8NnmdsBgoPipQD3aPDiLZZPK6l7xCFp8ZECZbOE6TnVWRQ5d9hpoWU6fPQxCmvXOHG0yQjnpBbUFcTVnmLIgKYp_MMpyC98ktGgml-3rke-U2SAGxZ_LvB8DSxLE1KkD9hE5voTWZUnA5LHB1fNAbw8DB8v_jVO8Ng
Application name: Monaco Editor
Windows 11 Enterprise (Version 24H2)  
OS build :26100.4770  

### Repro Steps:

1. Hit the URL: https://microsoft.github.io/monaco-editor/playground.html?source=v0.52.2#example-creating-the-editor-hello-world.
2. Monaco editor page will appear.
3. Apply high contrast.
4. Observe that In high contrast aquatic or desert modes, cursor is not visible at 'plain text'.

### Actual Results:
In high contrast aquatic or desert modes, cursor is not visible at 'plain text'.

### Expected Results:
In  'Aquatic & desert' mode, the contrast ratio for cursor present in 'plain text' should be clearly visible to the users,

### User Impact:
Low vision users cannot see where the cursor is, making it difficult to interact with or edit plain text fields.

### Attachment:

https://github.com/user-attachments/assets/c75768f3-dda6-4666-96b4-9d5960846939

<img width="1366" height="768" alt="Image" src="https://github.com/user-attachments/assets/74ac0870-91ef-4d95-b0d6-efd5c3161024" />

**Comment by AnKushSingh05** [comment]
Verified in the latest build 0.54.0-dev-20250924 issue is fixed

<img width="1280" height="720" alt="Image" src="https://github.com/user-attachments/assets/9a0d9b0a-72a2-4da5-9bb1-869135c36721" />

**Comment by meganrogge** [comment]
Yes, that is the fix
